[v1.4][W09-B4] Flashy

jethrokuan opened the pull request.

jethrokuan committed:
Update Developer Guide (#1)

- update NFRs
- update Target User Profile

jethrokuan committed:
Fix Travis URL (#2)

* Update travis URL

* Update link references in Developer Guide

jethrokuan committed:
Add Card model (#16)

* Add Card model

* Use front and back instead of question and answer

yong-jie committed:
Add theme changing functionality (#18)

* Add theme changing feature
* Change CSS loading implementation in MainWindow class

jethrokuan committed:
Add Card Book (#19)

* Add UniqueCardList

* Add CardBook

* fix style errors

* add newlines

* fix style errors

jethrokuan committed:
fix MESSAGE_USAGE in ChangeThemeCommand

jethrokuan committed:
move current theme into UiManager

This is because the current theme affects components beyond the MainWindow.

yong-jie committed:
Add quit command alias (#28)

shawnclq committed:
Add an add card feature (#21)

* Add an add card feature

* Minor changes

* Fix style issues

* Fix tests issues

* Fix style errors

yong-jie committed:
Feat/user-guide (#37)

* Add help prompts for adding card and changing theme

* Add help prompt for quitting the application

* Update command summary

* Fix typo error

* Update introduction paragraph

jethrokuan committed:
Add list card command (#38)

* Add UUID Predicate for Card

* Add UUID to card class

* Add list card command

jethrokuan committed:
Add stub UI for JavaFX (#42)

yong-jie committed:
Update documentation (#43)

* Update README

jethrokuan committed:
Update about us with Jethro's image (#45)

jethrokuan committed:
update userdoc to include list command (#44)

jethrokuan committed:
Add parser command tests for `addc` and `listc` (#50)

* add test for addCardCommand

* add parser test for ListCardCommand

* fix tests

shawnclq committed:
Update AboutUs.adoc (#51)

jethrokuan committed:
Add UI tests for Card-related functionality (#48)

* Add GUI tests for CardCard

* add GUI tests for CardListPanel

* fix style errors

yong-jie committed:
Add tests for theme changing functionality (#39)

* Add tests for theme changing functionality

* Fix style issues

* Refactor MESSAGE_INVALID_THEME logic message

* Refactor parseTheme method in ParserUtil

jethrokuan committed:
Remove tag functionality from addressBook (#55)

* Remove tag functionality from addressBook

* fix test

jethrokuan committed:
Refactor Person to Tag (#56)

* Refactor Person to Tag

- BrowserPanel related items have been left untouched, because they
  will probably be deleted
- User and Developer Guide not updated

* fix style error

yong-jie committed:
Autosave current theme in user preferences (#59)

jethrokuan committed:
Edit Tag model to contain correct fields (#57)

* Remove email field from Tag

* Remove phone field from Tag

* Address -> Description

* update test items to proper tags

* increase timeout for GUI bot to prevent test run failure

* add more address -> description cleanup

* fix style errors

* one more!

* add more address -> description

jethrokuan committed:
Remove CardBook (#62)

jethrokuan committed:
Add CardTag class (#64)

* update guava version

* add CardTag class

* Delete that distracting blank line in Card.java

yong-jie committed:
Move CardListPanel to a new column (#65)

* Edit UI to match mockup

shawnclq committed:
Add delete card feature (#66)

jethrokuan committed:
Add ID to Card and Tag (#68)

* make CardTag a non-singleton

* Require uuid string to be equal

* fix failing tests

* Add tests for XmlAdaptedCard

* add ID to Tag, and fix broken tests

* fix style errors

yong-jie committed:
Refactor UI events in preparation of card-back viewing functionality (#69)

* Refactor events in UI in preparation for back-viewing functionality

* Fix style errors

* Fix more style errors

* Nuke tests that check browserpanel

* Remove unused imports

* Fix style errors

* Rename event classes for clarity

TomatoCream committed:
Update AboutUs.adoc (#70)

* Update AboutUs.adoc

* Update AboutUs.adoc changed from jpg to png

jethrokuan committed:
Fix Travis and Coveralls badge (#71)

goweiwen commented on docs/DeveloperGuide.adoc:
```diff
@@ -10,9 +10,9 @@ ifdef::env-github[]
+:repoURL: https://github.com/CS2103JAN2018-W09-B4/addressbook-level4/tree/master
 
-By: `Team SE-EDU`      Since: `Jun 2016`      Licence: `MIT`
+By: `W09-B4`      Since: `Jun 2016`      Licence: `MIT`
```
The heading in the DG should be updated as well.

goweiwen commented on docs/DeveloperGuide.adoc:
```diff
@@ -781,14 +781,11 @@ See this https://github.com/se-edu/addressbook-level4/pull/599[PR] for the step-
 == Product Scope
 
 *Target user profile*:
+* has the need for flashcards
```
This section isn't formatted properly when rendered on your GitHub pages.

goweiwen commented on docs/DeveloperGuide.adoc:
```diff
@@ -781,14 +781,11 @@ See this https://github.com/se-edu/addressbook-level4/pull/599[PR] for the step-
+*Value proposition*: helps user memorise facts better.
 
 [appendix]
 == User Stories
```
Please include the user stories and use cases that you showed me in week 7.

goweiwen commented on docs/UserGuide.adoc:
```diff
@@ -60,6 +60,15 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
+* `theme t/light`
+* `theme t/dark`
+
 === Adding a person: `add`
```
Is this feature still in? Remember to update your User Guide for every change to reduce the chances of any stale information misleading users.

goweiwen reviewed (commented):
Please leave this PR open (do not close it) as we will use it to review your change until v1.5.

**Important**: As mentioned in the v1.0 review, please add your user-stories and use cases to the developer guide by the next tutorial so I can review them. Until then, I will consider this week's team LO as incomplete.

Keep in mind that since you are not merging your minor changes from v1.0, you will need to add new changes and let me know which feature you wish to consider your minor enhancement.

Please look through the user guide and developer guide together and update the relevant sections. For this week, you are required to think through the product's end behaviour and write the entire UG, for **all** the commands that you want to implement.

goweiwen commented on src/main/java/seedu/address/logic/commands/ExitCommand.java:
```diff
@@ -8,7 +8,8 @@
 
-    public static final String COMMAND_WORD = "exit";
+    public static final String COMMAND_WORD_EXIT = "exit";
+    public static final String COMMAND_WORD_QUIT = "quit";
```
You should not name your constants to match their content, as this makes them quite redundant and is not resistant to change. Instead, do something like `COMMAND_WORD` and `COMMAND_WORD_ALIAS`.

goweiwen commented on docs/UserGuide.adoc:
```diff
@@ -60,6 +60,15 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
 
 Format: `help`
 
+=== Changing the theme : `theme`
```
Good job implementing this enhancement 👍.

goweiwen commented on docs/UserGuide.adoc:
```diff
@@ -73,11 +82,24 @@ Examples:
 * `add n/John Doe p/98765432 e/johnd@example.com a/John street, block 123, #01-01`
 * `add n/Betsy Crowe t/friend e/betsycrowe@example.com a/Newgate Prison p/1234567 t/criminal`
 
+=== Adding a flashcard: `addc`
```
Well done for this feature as well 👍.

goweiwen reviewed (commented):
A few comments on the minor enhancements. Your code base is quite well tested, keep up the good work 👍.

jethrokuan committed:
Add CardTag to AddressBook (#75)

* Add XmlAdaptedCardTagEdge

* update CardTag to simpler format

* Add XmlAdaptedCardTag with tests

* Fix style errors

shawnclq committed:
Update docs to reflect major and minor enhancements (#76)

* Update DG and UG

* Update DG

jethrokuan committed:
Do not setCardTag from serialized addressbook if it does not exist (#78)

jethrokuan committed:
Update edit command and tests to ensure UUID has no change (#79)

shawnclq committed:
Edit card feature (#73)

* Add edit card feature

* Fix edit card issues

* Fix edit card test issues

shawnclq committed:
Update DG (#80)

* Update DG

* Update UG

* Update DG

* Fix style errors

TomatoCream committed:
Add Schedule for Flashcard Review (#82)

goweiwen commented on docs/DeveloperGuide.adoc:
```diff
@@ -358,6 +358,16 @@ image::UndoRedoActivityDiagram.png[width="650"]
+** Cons: No ability to customize the look and feel of the application.
+* **Alternative 2**: Support multiple themes.
+** Pros: Users get a certain extent of customizability in the application.
+** Cons: More development effort needed to support multiple themes and ensure that they can be loaded.
```
This sounds like simply considering implementing the feature or not. Instead, it should be a consideration faced during implementation. For example, customizable vs preset, or fully themeable vs just changing colours.

goweiwen commented on docs/DeveloperGuide.adoc:
```diff
@@ -493,340 +503,226 @@ Here are the steps to create a new release.
 
-*Scenario:* You are a software maintainer for `addressbook`, as the former developer team has moved on to new projects. The current users of your application have a list of new feature requests that they hope the software will eventually have. The most popular request is to allow adding additional comments/notes about a particular contact, by providing a flexible `remark` field for each contact,  ...
+[appendix]
+== Product Scope
```
Duplicate section?

goweiwen commented on docs/DeveloperGuide.adoc:
```diff
@@ -493,340 +503,226 @@ Here are the steps to create a new release.
-==== Step-by-step Instructions
+=== Jethro Kuan
+Major enhancement: Implementing a graph structure to link Cards and Tags so that users are able to tag different cards
+in order to search them efficiently.
```
How about putting the end goal in front? The implementation is just a minor detail.

goweiwen commented on docs/DeveloperGuide.adoc:
```diff
@@ -493,340 +503,226 @@ Here are the steps to create a new release.
+=== Foo Yong Jie
+Major enhancement: Improving the UI / UX of the application, including implementing separate panels for tags, cards,
+and the display of cards.
+This allows users to use the application efficiently.
```
This major enhancement is a bit vague. Maybe go into detail on what are the main changes you aim to make?

goweiwen commented on docs/DeveloperGuide.adoc:
```diff
@@ -493,340 +503,226 @@ Here are the steps to create a new release.
+[width="100%",cols="22%,<23%,<25%,<30%",options="header",]
+|=======================================================================
+|Priority |As a ... |I want to ... |So that I can...
+| `* *` | University Student | Create a new card                                        | I can review new material
```
No "must have" user stories? I feel like adding and reviewing cards at least are quite important in a flash card application...

goweiwen commented on docs/DeveloperGuide.adoc:
```diff
@@ -358,6 +358,16 @@ image::UndoRedoActivityDiagram.png[width="650"]
 ** Cons: Requires dealing with commands that have already been undone: We must remember to skip these commands. Violates Single Responsibility Principle and Separation of Concerns as `HistoryManager` now needs to do two * different things.
 // end::undoredo[]
 
+===== Aspect: Support for multiple themes
```
This is a subsection of the Undo/Redo feature. It should be part of a new section describing the implementation of multiple themes.

goweiwen reviewed (commented):
Good job! Your team looks on track to finishing your planned enhancements.

Make sure your user guide is up to date and ready before user testing happens.

Consider adding some documentation to your developer guide as well (specifically the Implementation section).

yong-jie committed:
Add ability to filter cards based on selected tag (#94)

* Add ability to filter card view upon clicking a tag

* Fix style errors

* Fix consistency issues

* Return filtering api for tests to run properly

* Remove blank lines

* Fix UniqueCardList test case

* Fix bug in ModelManager where setAll() affects addressbook

* Fix test cases

* Remove unused imports

* Remove unused command in AddCommandTest

* Remove trailing whitespace in AddCommandTest

jethrokuan committed:
Remove tag description (#95)

TomatoCream committed:
Improved Scheduling algorithm, Increased Intervals

shawnclq committed:
Update DG with card model and card storage documentation (#99)

TomatoCream committed:
Updated Developer Guide, included Schedule class (#100)

jethrokuan committed:
Update add tag command to not throw DuplicateTagException (#97)

even if duplicate.

TomatoCream committed:
Updated Developer Guide: easingFactor added to Schedule Documentation (#102)

* Updated Developer Guide easingFactor added to Schedule Documentation

* Updated User Guide added schedule documentation

yong-jie committed:
Add image and documentation for change theme command to DG (#101)

* Add Image and documentation to DG

* Fix title

jethrokuan committed:
Add documentation for CardTag Implementation (#98)

shawnclq committed:
Add MCQ-type card (#104)

* Add MCQ-type card

* Update UG

TomatoCream committed:
Improve introduction in User Guide (#103)

yong-jie committed:
Fix typo in UG and add more elaboration (#106)

* Fixed typo in ug and add more elaboration

* Organise organize

jethrokuan committed:
Allow adding of tags through addCardCommand (#109)

* Update add card command to add tags too

* Write tests for AddCardCommandParser

TomatoCream committed:
add showDue command (#108)

* Add showDue command and tests

* Add selectedCard to modelManager

TomatoCream committed:
add setRelativeNextReview (#105)

shawnclq committed:
Add card types (#111)

* Add MCQ-type card

* Update UG

* Add various tests

jethrokuan committed:
Allow changing tag set with EditTagCommand (#112)

* Use Optional for tags in AddCardCommandParser

* Allow edit command to add tags

* resolve merge conflict

yong-jie committed:
Fix bug where cardlist unfilters itself on add/del card (#120)

yong-jie committed:
Add command to select card (#119)

jethrokuan committed:
add ability to add and remove tags in editCardCommand (#122)

* add ability to add and remove tags in editCardCommand

* Add one more test case for tag not found

* fix typo

* Fix test case from SelectCardCommandSystemTest

yong-jie committed:
Fix clear command bug (#124)

* Fix issue updating issue in ModelManager

* Fix test cases and clear command bug

* Fix style

* Change permission of updateFilteredCardList to private

jethrokuan committed:
Add system test for AddCardCommand (#123)

* Add system test for AddCardCommand

* fix style errors

jethrokuan committed:
Refactor CardTag implementation (#127)

* Simplified Model interface for CardTag

* Add more documentation for CardTag

shawnclq committed:
Add blanks card (#128)

* Change HashSet to LinkedHashSet to preserve order of options

* Add fill-in-blanks cards, add tests, changed Set to List for MCQ cards and fill blanks cards

* Modify Fill Blanks cards, remove need for multiple FRONT prefixes

jethrokuan committed:
Delete Tag and Card now triggers CardMap update (#130)

* update delete tag to update CardTag

* update delete card to update CardMap

yong-jie committed:
Fix auto-updating of card list when edit card is used (#131)

yong-jie committed:
Add autocompletion feature (#132)

* Add autocomplete functionality

* Fix documentation

* Add more autocomplete text

* Add more autocomplete text and fix addressbook parser test

* Fix style errors

* Add final batch of autocomplete texts

TomatoCream committed:
add Answer, Showdue(Date), Schedule(Date). (#129)

* Add showDue command and tests

* Add selectedCard to modelManager

* Add answerCommand

* Add indefinate scheduling for showDue

* Add date specific scheduling for schedule

* add ScheduleCommandParserTest

* add ShowDueCommandParserTest

* add ScheduleCommandTest

* add ModelManagerTest

* add ScheduleTest

* add AnswerCommandParserTest

* add ScheduleCommandTest

* add AnswerCommandTest

* add ScheduleTest

* add ShowDueTest

* add AddressBook getReviewList test

yong-jie committed:
Add card-back viewing functionality (#133)

* Add card back ui panel

* Replace browser panel with card back completely

* Fix style issues

* Fix test cases that require browserpanel

jethrokuan committed:
Merge functionality of list command (#134)

shawnclq committed:
Add edit flashcard feature (#135)

* Add edit flashcard feature

* Fix style issues

* Remove updateFilteredCard()

jethrokuan committed:
Remove add command (#136)

TomatoCream committed:
Reorder user guide

shawnclq committed:
Update UG (#139)

jethrokuan committed:
remove tags when card is removed under conditions (#138)

when cards are removed, some tags no longer have cards related to
them. Hence, they can be safely removed.

shawnclq committed:
Add tests for edit card (#146)

jethrokuan committed:
Fix user docs for edit tag command (#168)

jethrokuan committed:
Add list command to user docs (#170)

shawnclq committed:
Update UG to exclude editm command (#171)

jethrokuan committed:
Update title of application to "Flashy" (#169)

* update shadowJar build task to produce `flashy.jar`

* update app title to Flashy

jethrokuan committed:
Update sample data (#173)

tags now have at least one card, each card has at least one tag.

TomatoCream committed:
update Userguide, Showdue, answer, schedule (#172)

jethrokuan committed:
Show tags in card (#175)

yong-jie committed:
Update User Guide (#176)

* Fix grammatical error in 4.3, and fill in autocomplete function

* Rename misc to qol

* Enhance autocompletion documentation

* Fix autocompletion docs

* Enhance clearing of data and history

* Fix undo/redo and exit doc

* Fix typo

* Fix typo in saving data

* Import changes

jethrokuan committed:
Add flag to list command to filter untagged cards (#177)

* Add flag to list that allows listing of cards without tags

* update user guide for list command

* fix style errors

TomatoCream committed:
showDue now filters on top of filteredCards (#178)

TomatoCream committed:
Show due message (#179)

* showDue now filters on top of filteredCards

* add showDue when empty message

yong-jie committed:
Add functionality to view MCQ options in card list card (#181)

* Add functionality to display options in MCQ cards

* Fix style

* Add collation

TomatoCream committed:
Flexible date parser (#180)

* showDue now filters on top of filteredCards

* date parser can now handle multiple arguments

* flexible parsing of dates

TomatoCream committed:
Add stem latexmath to developer guide (#174)

yong-jie committed:
Add empty card back event (#182)

TomatoCream committed:
showDue will clear (#183)

yong-jie committed:
Fix bug where cardback doesnt empty (#184)

shawnclq committed:
Merge XmlAdaptedMcqCard into XmlAdaptedCard (#185)

yong-jie committed:
Enhance light theme (#187)

* Add functionality to display options in MCQ cards

* Fix style

* Enhance theme

* Fix tests

jethrokuan committed:
Fix list command message (#189)

Give more meaningful message if flag is passed

jethrokuan committed:
Fix AddCardCommand message (#190)

addressbook -> cardbank

yong-jie committed:
Fix bug where select would not clear the card back (#188)

* Fixed bug where select would not clear the card back

* Fix test

jethrokuan committed:
Fix Find command in User Guide (#191)

TomatoCream committed:
Project Portfolio Ding Feng (#193)

jethrokuan committed:
Add initial draft of jethrokuan's developer portfolio (#192)

shawnclq committed:
Add first draft for developer's portfolio

* Add first draft for developer's portfolio

* Edit DG

shawnclq committed:
Tweak UI to show option value for back for card (#195)

* Tweak UI to show option value for back for card and show numbered list for options, and add EditCardCommandSystemTest

* Improve CardCard tests

shawnclq committed:
Change diagrams to black and white (#196)

yong-jie committed:
Add first draft of portfolio (#197)

* Add selectc to user guide

* Add portfolio

* Fix style errors

shawnclq committed:
Clean up DG diagrams and add in current implementation for polymorphic cards and storage (#200)

jethrokuan committed:
Update developer guide for `CardTag` (#201)

* Initial update

* update dev guide

jethrokuan committed:
Add logging to CardTag class (#202)

yong-jie committed:
Enhance light theme (#203)

* Fix colours

* Fix card back

* Update ui image

* Change to default light theme

* Fix typo

* Update portfolio and UGDG

* Add collation markers

TomatoCream committed:
Fix showdue bug (#199)

yong-jie committed:
Enhance dark theme (#204)

* Change outdated comment

* Enhance dark theme

yong-jie committed:
Add more documentation to the developer guide (#206)

* Update developer guide for themes

* Add swimlane diagram

* Add autocomplete docs

TomatoCream committed:
Update UserGuide, add preface, reorder some, add logo (#198)

* UserGuide update. add preface, reorder some, add logo

* Clean up on Logo

shawnclq committed:
Add logging (#207)

jethrokuan committed:
Update User Guide (#208)

* Update User Guide

* fix based on @fooyongjie's comments

TomatoCream committed:
refactor date tests (#205)

* refractor date tests

* refactor date test

shawnclq committed:
Fix sample data (#209)

yong-jie committed:
Collate code (#210)

TomatoCream committed:
Add portfolio for Ding Feng (#211)

shawnclq committed:
Rename address to flashy (#213)

* Rename seedu.address to seedu.flashy

* Rename AddressBook

yong-jie committed:
Revert "Rename address to flashy (#213)" (#215)

This reverts commit 57b5dfef5b4b40ae15e7f15e975cdaa11920ea5a.

jethrokuan committed:
Update user portfolio (#212)

TomatoCream committed:
Project portfolio update Ding Feng (#214)

TomatoCream committed:
Update Command summary to sentence case (#217)

* update userguide summary showdue, answer, schedule

* change summary to sentence case

TomatoCream committed:
Refactor Answer Test (#216)

shawnclq committed:
Tweak 'addc' section of UG, add authorship (#219)

* Tweak 'addc' section of UG, add authorship

* Fix issues

TomatoCream committed:
update project portfolio ding feng (#220)

TomatoCream committed:
Collated final (#222)

shawnclq committed:
Edit documents (#223)

yong-jie committed:
Add FAQ and renamed addressbook to flashy in documentation (#224)

* Populate FAQ

* Modify ppp

* Modify enhancements in dev guide

* Fix formatting

* Rename to flashy

* Modify user guide

* Remove capitalisation of flashy

shawnclq committed:
Rename (#221)

* Rename seedu.address to seedu.flashy

* Rename AddressBook

* Edit documentation error

yong-jie committed:
Remove encryption (#225)

* Remove encryption

* Collate code
